draft-irtf-cfrg-aegis-aead.md

```diff
@@ -1620,6 +1622,191 @@ Instead of relying on the generic `Encrypt` function, implementations can skip t
 
 After initialization, the `Update` function is called with constant parameters, allowing further optimizations.
 
+# AEGIS as a Message Authentication Code
+
+All AEGIS variants can be used to construct a MAC.
+
+For all the variants, the `Mac` function takes a key, a nonce, and data as input, and produces a 128- or 256-bit tag as output.
+
+This is the only function that allows the reuse of `(key, nonce)` pairs with different inputs.
+
+AEGIS-based MAC functions MUST NOT be used as hash functions. If the key is known, inputs causing state collisions can easily be crafted.
+Likewise, unlike hash-based MACs, tags MUST NOT be used for key derivation, as there is no guarantee that they are uniformly random.
+
+~~~
+Mac(data, key, nonce)
+~~~
+
+Inputs:
+
+- `data`: the input data to authenticate (length MUST be less than or equal to `A_MAX`).
+- `key`: the encryption key.
+- `nonce`: the public nonce.
+
+Outputs:
+
+- `tag`: the authentication tag.
+
+## AEGISMAC-128L
+
+AEGISMAC-128L refers to the `Mac` function based on the building blocks of AEGIS-128L.
+
+Steps:
+
+~~~
+Init(key, nonce)
+data_blocks = Split(ZeroPad(data, 256), 256)
+for di in data_blocks:
+    Absorb(di)
+tag = Finalize(|data|, tag_length) # tag_length is 16 or 32
+return tag
+~~~
+
+## AEGISMAC-256
+
+Similarly, AEGISMAC-256 refers to the `Mac` function based on the building blocks of AEGIS-256.
```

Comment:
I wonder if we should remove the `Similarly` since we're not doing it between AEGISMAC-128X and -256X.